tests/autotuner/test_autotuner_core.py

```
@@ -1015,3 +1015,71 @@ def test_skip_ops_restored_after_context():
     assert tuner._effective_skip_ops == frozenset()
+
+
+def test_find_nearest_profile_cache_stable_with_reused_callable():
+    """Reusing the same map_to_tuning_buckets callable should not grow the lru_cache.
+
+    This verifies the fix for the unbounded lru_cache leak: when the same lambda
+    object is reused (via _get_map_to_hybrid_bucket), id() stays stable, so
+    DynamicTensorSpec.__hash__ produces the same hash, and _find_nearest_profile's
+    lru_cache hits instead of creating a new entry every call.
+    """
+    map_fn = lambda x: min(x, 8192)
```

Ruff (0.15.20)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `map_fn` as a `def`

(E731)